lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -890,6 +934,17 @@ def __init__(
             RelationRoleMismatchError: If the relation with the same name as provided
                 via `relation_name` argument does not have the `RelationRole.provides`
                 role.
+
+        Emits:
+            loki_push_api_endpoint_joined: This event is emitted when the relation between the
+                charmed operator that implements `LokiPushApiProvider` (Loki charm for instance)
+                and the charmed operator that implements `LokiPushApiConsumer` is established.
+            loki_push_api_endpoint_departed: This event is emitted when the relation between the
+                charmed operator that implements `LokiPushApiProvider` (Loki charm for instance)
+                and the charmed operator that implements `LokiPushApiConsumer` is removed.
+            loki_push_api_alert_rules_error: This event is emitted whether alert rules files does
+                not contain the keys `alert` or `expr` or there is no alert rules file in
+                `alert_rules_path`.
```

Comment:
```suggestion
            loki_push_api_alert_rules_error: This event is emitted when an invalid alert rules
                file is encountered or if `alert_rules_path` is empty.
```

Reply:
Is it emitted every time there's a lifecycle event or just once?

Reply:
It's only on `relation_changed` that this is checked, unless I missed something.

We don't necessarily have to keep exactly the same pattern, but for comparison, the Grafana Dashboard events I mentioned yesterday have two properties. One is a `bool` indicating whether or not it's valid, and the other is the actual message.

[Usage is here](https://github.com/canonical/grafana-operator/blob/main/lib/charms/grafana_k8s/v0/grafana_dashboard.py#L447), with the actual event definition [here](https://github.com/canonical/grafana-operator/blob/e3b5795cb3ff0da328549b3e266bfb8c1ebb4047/lib/charms/grafana_k8s/v0/grafana_dashboard.py#L249).

It potentially gives a little more flexibility in the future. Not even remotely blocking on this, just a note/example of adding more properties to events.

Reply:
@rbarry82 You are right, we are using only on `relation_changed`.

The Grafana Dashboard event you mentioned it's a very good example. I'll keep in mind, thanks!!